### DISCUSSION TITLE
Inability to Generate a code signing certificate for free users

### DISCUSSION THREAD

**Discussion by adamappdev** [codemagic-ci-cd/codemagic-docs#929] [Q&A] [answered]
Hi there,

We're experimenting with CodeMagic for our Flutter iOS/Android builds in my organization. As we have not reached to a successful build yet, we have not decided to move to the paid version yet, so we're still using the free version.

Our current problem in our builds is the `flutter build ipa` command to build the iOS archive. It completes with this log:
```
Building without sound null safety
For more information see https://dart.dev/null-safety/unsound-null-safety

Building App Store IPA...                                           3.5s
Encountered error while creating the IPA:
error: exportArchive: "Runner.app" requires a provisioning profile.

Try distributing the app in Xcode: "open /Users/builder/clone/build/ios/archive/Runner.xcarchive"
```

I think this does have something to do with what's described [in this doc](https://docs.codemagic.io/yaml-code-signing/signing-ios/). However, In the [Adding the Code signing certificate step](https://docs.codemagic.io/yaml-code-signing/signing-ios/#adding-the-code-signing-certificate), it instructs me to `Open your Codemagic Team settings`, and as I am in the free tier I do not have that section in my console.

**Comment by icarusdust** [comment]
Hi @adamappdev, are you uploading your provisioning profile via code signing identities ? If so, is it matching your uploaded certificate ? A build id would be great to see it in more details.

**Reply by adamappdev** [reply]
Thanks for such a fast reply!

I did not have a step where I did something with provisioning profile. I configured my code signing to be automatic from Codemagic. I may have missed a step maybe, I'm not sure what you are referring to.

This is my latest build id `631087434aad2afe1d75ac1e`, it is marked as successful but the `flutter build ipa` step returns as successful even though it throws an error.
Here is a screenshot of the output of the build: (I've replaced my secret ids with asterisks)
![image](https://user-images.githubusercontent.com/110532071/187899826-4d88a152-e844-41a4-a45f-e23012c7bd1f.png)

**Comment by icarusdust** [comment, marked as answer]
Thanks for sharing the build id. The documentation reference above referred to signing identities, hence why my question about uploading it manually. 
Let's start with your yaml first, I see that your android settings exist in your ios workflow as well. I am assuming that you downloaded the ready yaml from the Workflow Editor. Please try to configure your yaml as [this](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/flutter/flutter-android-and-ios-yaml-demo-project/codemagic.yaml#L73) and let's continue from that point on.

**Reply by adamappdev** [reply]
Thanks, I'll start with that and notify you when I'm done!

**Reply by adamappdev** [reply]
Hi @icarusdust,

I've filled my new `codemagic.yaml` file according to the link you have sent me, except for `certificate_credentials`, is this supposed to be filled by following [this doc](https://docs.codemagic.io/yaml-code-signing/signing-ios/#adding-the-code-signing-certificate)?

**Reply by icarusdust** [reply]
Yes, it is necessary to provide it with automatic code signing option. Please check [this doc](https://docs.codemagic.io/yaml-code-signing/code-signing-personal/) instead.

**Reply by icarusdust** [reply]
By certificate_credentials, I believe you mean CERTIFICATE_PRIVATE_KEY.

**Reply by adamappdev** [reply]
As the [template file](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/flutter/flutter-android-and-ios-yaml-demo-project/codemagic.yaml#L73) mentions `certificate_credentials`, what I'll be doing is to insert my `CERTIFICATE_PRIVATE_KEY` into the console `Environment variables` and add it into a group with the name `certificate_credentials`, is that correct?

**Reply by adamappdev** [reply]
I've overcome the problem of this discussion, now I'm having another one in the `Publishing` step, but I think I'll figure this out on my own. @icarusdust Thanks for your efforts and very fast replies.
